baseband/vlbi_base/tests/test_vlbi_base.py

```diff
@@ -425,34 +425,121 @@ def test_vlbi_file_base(self, tmpdir):
+                                   bps=1, complex_data=False, subset=subset,
+                                   unsliced_shape=unsliced_shape,
+                                   samples_per_frame=1000,
+                                   sample_rate=10000*u.Hz, squeeze=False)
+
+    def test_sample_shape_and_squeeze(self):
+        # Tests stream base's sample and squeezing routines.
+
+        smp_shp_mkr = namedtuple('SampleShape',
+                                 'n0, n1, n2, n3, n4, n5, n6, n7')
+        unsliced_shape = (1, 17, 3, 2, 1, 5, 1, 1)
+
+        # Try tuple only.
         sb = VLBIStreamBase(fh_raw=None, header0=None,
-                            sample_shape=sample_shape, bps=1,
-                            complex_data=False, thread_ids=None,
+                            bps=1, complex_data=False, subset=None,
+                            unsliced_shape=unsliced_shape,
                             samples_per_frame=1000,
                             sample_rate=10000*u.Hz, squeeze=False)
-        assert sb.sample_shape == sample_shape
+        assert sb.sample_shape == unsliced_shape
+        sb.squeeze = True
+        assert sb.sample_shape == (17, 3, 2, 5)
+
+        # Try with equivalent sample shape.
+        sb = self.make_streambase_with_shape(smp_shp_mkr, unsliced_shape, None)
+        assert sb.sample_shape == unsliced_shape
         sb.squeeze = True
         assert sb.sample_shape == (17, 3, 2, 5)
-        data = np.empty((100, 17, 3, 2, 5), dtype='float32')
-        assert sb._unsqueeze(data).shape[1:] == sample_shape
-        smp_shp_cls_s = namedtuple('SampleShape',
-                                   'n1')
-        sample_shape_short = smp_shp_cls_s(1)
-        sbs = VLBIStreamBase(fh_raw=None, header0=None,
-                             sample_shape=sample_shape_short, bps=1,
-                             complex_data=False, thread_ids=None,
-                             samples_per_frame=1000, sample_rate=10000*u.Hz,
-                             squeeze=False)
-        assert sbs.sample_shape == sample_shape_short
+        assert sb.sample_shape._fields == ('n1', 'n2', 'n3', 'n5')
+
+        data = np.empty((100,) + sb.sample_shape, dtype='float32')
+        assert sb._unsqueeze(data).shape[1:] == unsliced_shape
+
+        # Check that single-axis sample shape squeezes to ().
+        smp_shp_mkr_s = namedtuple('SampleShape',
+                                   'n0')
+        unsliced_shape_short = (1,)
+        sbs = self.make_streambase_with_shape(smp_shp_mkr_s,
+                                              unsliced_shape_short, None)
+        assert sbs.sample_shape == unsliced_shape_short
         sbs.squeeze = True
         assert sbs.sample_shape == ()
         data = np.empty(100, dtype='float32')
-        assert sbs._unsqueeze(data).shape[1:] == sample_shape_short
+        assert sbs._unsqueeze(data).shape[1:] == unsliced_shape_short
+
+    @pytest.mark.parametrize(('subset', 'sliced_shape',
```

Comment:
You can use two @parametrize to get the "product" or different ones (or `itertools.product`)